tfhe/src/c_api/high_level_api/integers.rs

```diff
@@ -659,3 +659,112 @@ pub unsafe extern "C" fn compact_fhe_int256_list_try_encrypt_with_compact_public
+                seed_low_bytes: u64,
+                seed_high_bytes: u64,
+                random_bits_count: u64,
+            ) -> c_int {
+
+                $crate::c_api::utils::catch_panic(|| {
+                    let seed_low_bytes: u128 = seed_low_bytes.into();
+                    let seed_high_bytes: u128 = seed_high_bytes.into();
+                    let seed = concrete_csprng::seeders::Seed((seed_high_bytes << 64) | seed_low_bytes);
+
+                    let result = crate::high_level_api::par_generate_oblivious_pseudo_random_unsigned_integer(seed, random_bits_count);
+                    *out_result = Box::into_raw(Box::new($name(result)));
+                })
+            }
+        }
+    };
+}
+
+macro_rules! impl_oprf_for_int {
+    (
+        name: $name:ident
+    ) => {
+
+        ::paste::paste! {
+            #[no_mangle]
+            pub unsafe extern "C" fn [<par_generate_oblivious_pseudo_random_unsigned_ $name:snake>](
+                out_result: *mut *mut $name,
+                seed_low_bytes: u64,
+                seed_high_bytes: u64,
+                random_bits_count: u64,
+            ) -> c_int {
+                $crate::c_api::utils::catch_panic(|| {
+                    let seed_low_bytes: u128 = seed_low_bytes.into();
+                    let seed_high_bytes: u128 = seed_high_bytes.into();
+                    let seed = concrete_csprng::seeders::Seed((seed_high_bytes << 64) | seed_low_bytes);
+
+                    let result =
+                        crate::high_level_api::par_generate_oblivious_pseudo_random_signed_integer(
+                            seed,
+                            crate::high_level_api::SignedRandomizationSpec::Unsigned {
+                                random_bits_count
+                            },
+                        );
+                    *out_result = Box::into_raw(Box::new($name(result)));
+                })
+            }
+        }
+
+        ::paste::paste! {
+            #[no_mangle]
+            pub unsafe extern "C" fn [<par_generate_oblivious_pseudo_random_full_signed_ $name:snake>](
+                out_result: *mut *mut $name,
+                seed_low_bytes: u64,
+                seed_high_bytes: u64,
+            ) -> c_int {
+                $crate::c_api::utils::catch_panic(|| {
+                    let seed_low_bytes: u128 = seed_low_bytes.into();
+                    let seed_high_bytes: u128 = seed_high_bytes.into();
+                    let seed = concrete_csprng::seeders::Seed((seed_high_bytes << 64) | seed_low_bytes);
+
+                    let result = crate::high_level_api::par_generate_oblivious_pseudo_random_signed_integer(
+                        seed,
+                        crate::high_level_api::SignedRandomizationSpec::FullSigned,
+                    );
+                    *out_result = Box::into_raw(Box::new($name(result)));
+
+                })
+            }
+        }
+    };
+}
+
+impl_oprf_for_uint!(name: FheUint2);
```

Comment:
Maybe it changed but we used to have a macro to "instantiate" the functions for FheUint types, should this maybe be put inside that macro instead of here ?

Reply:
The macro `create_integer_wrapper_type` does not distinguish between unsigned and signed integers, so I don't think we can use it